pkg/hintrunner/zero/zerohint_keccak.go

```diff
@@ -68,6 +69,114 @@ func createCairoKeccakFinalizeHinter(resolver hintReferenceResolver) (hinter.Hin
 	return newCairoKeccakFinalizeHint(keccakPtrEnd), nil
 }
 
+// UnsafeKeccak computes keccak hash of the data in memory without validity enforcement and writes the result in the `low` and `high` memory cells
+//
+// `newUnsafeKeccakHint` takes 4 operanders as arguments
+//   - `data` is the address in memory to the base of the data array to hash is stored. Each word in the array is 16 bytes long, except the last one
+//   - `length` is the length of the data to hash
+//   - `low` is the low part of the produced hash
+//   - `high` is the high part of the produced hash
+func newUnsafeKeccakHint(data, length, high, low hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "UnsafeKeccak",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//>	from eth_hash.auto import keccak
+			//>	data, length = ids.data, ids.length
+			//>	if '__keccak_max_size' in globals():
+			//>		assert length <= __keccak_max_size, \
+			//>			f'unsafe_keccak() can only be used with length<={__keccak_max_size}. ' \
+			//>			f'Got: length={length}.'
+			//>	keccak_input = bytearray()
+			//>	for word_i, byte_i in enumerate(range(0, length, 16)):
+			//>		word = memory[data + word_i]
+			//>		n_bytes = min(16, length - byte_i)
+			//>		assert 0 <= word < 2 ** (8 * n_bytes)
+			//>		keccak_input += word.to_bytes(n_bytes, 'big')
+			//>	hashed = keccak(keccak_input)
+			//>	ids.high = int.from_bytes(hashed[:16], 'big')
+			//>	ids.low = int.from_bytes(hashed[16:32], 'big')
+
+			lengthVal, err := hinter.ResolveAsUint64(vm, length)
+			if err != nil {
+				return err
+			}
+			keccakMaxSize, err := ctx.ScopeManager.GetVariableValueAsUint64("__keccak_max_size")
+			if err != nil {
+				return err
+			}
+			if lengthVal > keccakMaxSize {
+				return fmt.Errorf("unsafe_keccak() can only be used with length<=%d.\n Got: length=%d.", keccakMaxSize, lengthVal)
+			}
```

Comment:
Put this check in `if err == nil` instead? Given its a conditional check.

Reply:
Technically you are right, but I assigned an arbitrary value of 2**20 to the keccakMaxSize (may be moved to globals in the future)